src/player.cpp

```diff
@@ -4121,24 +4121,24 @@ int player::overmap_sight_range(int light_level) const
     if( sight <= SEEX * 4) {
         return (sight / (SEEX / 2) );
     }
-    if ((has_amount("binoculars", 1) || has_amount("rifle_scope", 1) ||
-        has_amount("survivor_scope", 1) || -1 != weapon.has_gunmod("rifle_scope") ) &&
+    if ((has_item_with_flag("ZOOM") || worn_with_flag("ZOOM") || 
+        -1 != weapon.has_gunmod("rifle_scope") ) &&
         !has_trait("EAGLEEYED"))  {
          if (has_trait("BIRD_EYE")) {
              return 25;
          }
         return 20;
     }
-    else if (!(has_amount("binoculars", 1) || has_amount("rifle_scope", 1) ||
-        has_amount("survivor_scope", 1) || -1 != weapon.has_gunmod("rifle_scope") ) &&
+    else if (!(has_item_with_flag("ZOOM") || worn_with_flag("ZOOM") ||
+        -1 != weapon.has_gunmod("rifle_scope") ) &&
```

Comment:
`has_gunmod()` is iterating over all `vector<item> contents` which can be uset to check `contents[i].has_flag()` instead of the `contents[i].typeId()`. Probably best to add a new function `has_gunmod_with_flag()`.